src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    static int size;
+    private Entry<T> first;
+    private Entry<T> last;
+
+    private static class Entry<T> {
+        T element;
+        Entry<T> next;
+        Entry<T> prev;
+
+        public Entry(T element, Entry<T> next, Entry<T> prev) {
+            this.element = element;
+            this.next = next;
+            this.prev = prev;
+        }
+
+        public Entry(T element) {
+            this.element = element;
+        }
+    }
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
+    private void checkExeptionFilledElement(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    private void checkExeption(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    public void addInTheEnd(T value) {
+        Entry<T> newEntry = new Entry<>(value, null, last);
+        if (size == 0) {
+            first = newEntry;
+        } else {
+            newEntry.prev.next = newEntry;
+        }
+        last = newEntry;
+        ++size;
+    }
+
+    private void addByIndex(T value, int index) {
```

Comment:
we use this method only at one place
let's move all code from here to method add(T value, int index)